core/domain/exp_jobs_one_off_test.py

```diff
@@ -1834,3 +1838,289 @@ def test_no_action_is_performed_for_deleted_exploration(self):
+                        'file_size_bytes': 123,
+                        'needs_update': True,
+                        'duration_secs': 0.0
+                    }
+                },
+                'default_outcome': {}
+            }
+        }
+
+        state1.update_recorded_voiceovers(
+            state_domain.RecordedVoiceovers.from_dict(recorded_voiceovers_dict))
+        exp_services.save_new_exploration(self.user_a_id, exploration)
+
+        job_id = exp_jobs_one_off.VoiceoverDurationSecondsOneOffJob.create_new()
+        exp_jobs_one_off.VoiceoverDurationSecondsOneOffJob.enqueue(job_id)
+        with self.swap(logging, 'error', _mock_logging_function):
+            self.process_and_flush_pending_tasks()
+
+        relative_path = ('File %s/%s not found.' % (self._FILENAME_PREFIX,
+                                                    self.TEST_AUDIO_FILE_MP3))
+        self.assertEqual(
+            observed_log_messages,
+            ['MP3 audio file exception for file %s ,EXP_ID: %s,'
+             'STATE_NAME: %s,REASON: %s' % (
+                 self.TEST_AUDIO_FILE_MP3, self.VALID_EXP_ID, state_name,
+                 relative_path)])
+        actual_output = (
+            exp_jobs_one_off.VoiceoverDurationSecondsOneOffJob.get_output(
+                job_id)
+            )
+        expected_output = [('[u\'FAILED_NO_CHANGE\', [u\'EXP_ID: exp_id0, '
+                            'FAILED_DURATION_CHANGE: 1\']]')]
+        self.assertEqual(actual_output, expected_output)
+
+    def test_no_action_is_performed_for_deleted_exploration(self):
+        """Test that no action is performed on deleted explorations."""
+
+        exp_id = '100'
+        self.save_new_valid_exploration(exp_id, self.user_a_id)
+        exp_services.delete_exploration(self.user_a_id, exp_id)
+
+        run_job_for_deleted_exp(
+            self, exp_jobs_one_off.VoiceoverDurationSecondsOneOffJob)
+
+    def test_minimum_allowed_state_schema_version(self):
+        exploration = exp_domain.Exploration.create_default_exploration(
+            self.VALID_EXP_ID, title='Exploration Title 1', category='category')
+        exploration.add_states(['State1'])
+        # Test with the unaccpted state_schema_version
+        # for duration_secs which is less than 31.
+        exploration.states_schema_version = 30
+        state1 = exploration.states['State1']
+        recorded_voiceovers_dict = {
+            'voiceovers_mapping': {
+                'content': {
+                    'en': {
+                        'filename': self.TEST_AUDIO_FILE_MP3,
+                        'file_size_bytes': 123,
+                        'needs_update': True,
+                        'duration_secs': 0.0
+                    }
+                },
+                'default_outcome': {}
+            }
+        }
+
+        state1.update_recorded_voiceovers(
+            state_domain.RecordedVoiceovers.from_dict(recorded_voiceovers_dict))
+
+
```

Comment:
Ditto, drop one of these newlines. Do the same elsewhere.

Reply:
done